File: Olympiad/COCI/official/2007/contest6_solutions/magija.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 6 - Task MAGIJA
  Programming language Pascal
*)

program magija;

var
   karta      : array[1..200, 1..200] of char;
   r, s, a, b : longint;
   i, j       : longint;

begin
   readln(r, s);
   for i:=1 to r do begin
      for j:=1 to s do begin
         read(karta[i, j]);
         karta[2*r-i+1, j] := karta[i, j];
         karta[i, 2*s-j+1] := karta[i, j];
         karta[2*r-i+1, 2*s-j+1] := karta[i, j];
      end;
      readln;
   end;

   readln(a, b);
   if karta[a, b] = '#' then
      karta[a, b] := '.'
   else
      karta[a, b] := '#';

   for i:=1 to 2*r do begin
      for j:=1 to 2*s do
         write(karta[i, j]);
      writeln;
   end;
end.

File: Olympiad/COCI/official/2007/contest6_solutions/magija.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 6 - Task MAGIJA
  Programming language C++
*/

#include <cstdio>
using namespace std;

int main(){
   char karta[201][201];
   int r, s, a, b;

   scanf("%d%d", &r, &s);

   for (int i=0; i<r; i++){
      for (int j=0; j<s; j++){
         scanf(" %c", &karta[i][j]);
         karta[2*r-i-1][j] = karta[i][2*s-j-1] = karta[2*r-i-1][2*s-j-1] = karta[i][j];
      }
   }

   scanf("%d%d", &a, &b);
   karta[a-1][b-1] = karta[a-1][b-1] == '#' ? '.' : '#';

   for (int i=0; i<2*r; i++) {
      karta[i][2*s] = '\0';
      printf("%s\n", karta[i]);
   }

   return 0;
}
